src/pygama/dsp/_processors/fixed_time_pickoff.py

```diff
@@ -38,10 +39,9 @@ def fixed_time_pickoff(w_in, t_in, a_out):
     if np.isnan(w_in).any() or np.isnan(t_in):
         return
 
-    if np.floor(t_in) != t_in:
-        raise DSPFatal('The pick-off index must be an integer')
-
     if int(t_in) < 0 or int(t_in) >= len(w_in):
         return
 
-    a_out[0] = w_in[int(t_in)]
+    i = int(t_in)
+    w = t_in-i
+    a_out[0] = (1-2)*w_in[i] + w*w_in[i+1]
```

Comment:
What is `(1-2)` doing there? Should it be `(1-w)`?

Reply:
also, I think it would be clearer to rename t_in as i_in since it is an index, not a time.